rest_server/table_config.py

```diff
@@ -0,0 +1,301 @@
+"""Values for the table config."""
+
+
+from typing import Any, Dict, Final, List, Tuple, TypedDict, Union
+
+# local imports
+from keycloak_setup.icecube_setup import ICECUBE_INSTS  # type: ignore[import]
+
+ID = "_id"
+WBS_L2 = "WBS L2"
+WBS_L3 = "WBS L3"
+LABOR_CAT = "Labor Cat."
+US_NON_US = "US / Non-US"
+INSTITUTION = "Institution"
+_NAMES = "Names"
+_TASKS = "Tasks"
+SOURCE_OF_FUNDS_US_ONLY = "Source of Funds (U.S. Only)"
+FTE = "FTE"
+NSF_MO_CORE = "NSF M&O Core"
+NSF_BASE_GRANTS = "NSF Base Grants"
+US_IN_KIND = "US In-Kind"
+NON_US_IN_KIND = "Non-US In-Kind"
+GRAND_TOTAL = "Grand Total"
+TOTAL_COL = "Total Of?"
+
+
+US = "US"
+NON_US = "Non-US"
+
+
+class _ColumnConfigTypedDict(TypedDict, total=False):
+    """TypedDict for column configs."""
+
+    width: int
+    non_editable: bool
+    hidden: bool
+    options: List[str]
+    sort_order: int
+    conditional_parent: str
+    conditional_options: Dict[str, List[str]]
+    border_left: bool
+    on_the_fly: bool
+    funding_source: bool
+    numeric: bool
+
+
+_COLUMN_CONFIGS: Final[Dict[str, _ColumnConfigTypedDict]] = {
```

Comment:
It almost feels like this should come from a config file (yaml maybe?).

Reply:
I agree it seems underdeveloped. My goal is to grab this from Smartsheet, so front-end config can be done remotely. But I'm not certain this is needed/ would be wanted.

Reply:
Ah.  If this is going to come from Smartsheet, then I withdraw my complaint.